inventory_management/inventory_management/doctype/stock_entry/stock_entry.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) 2024, Viny Selopal and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+from frappe.query_builder.functions import Sum
+
+
+class MandatoryWarehouseMissing(frappe.ValidationError):
+    pass
+
+
+class InsufficientItems(frappe.ValidationError):
+    pass
+
+
+class StockEntry(Document):
+    def validate(self):
+        self.validate_balance_qty()
+        self.validate_mandatory_warehouses()
+
+    def validate_balance_qty(self):
+        if self.type == "Receive":
+            return
+
+        for item_row in self.stock_entry_items:
+            item_warehouse_balance = get_item_balance_for_warehouse(
+                item_row.source_warehouse, item_row.item, self.stock_entry_items
```

Comment:
```suggestion
                item_row.source_warehouse, item_row.item
```
The function is only accepting 2 parameters

<img width="1292" alt="image" src="https://github.com/vinyselopal/InventoryManagementFrappe/assets/24353136/28288a04-3797-4ad5-9a47-322f6a3e6c05">